frontend/src/components/VideoCardPage/VideoCardPage.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import Grid from '@material-ui/core/Grid';
+import TopBar from '../TopBar/TopBar';
+import UploadModal from '../UploadModal/UploadModal';
+import './styles.css';
+import VideoCardsByDate from '../VideoCardsByDate/VideoCardsByDate';
+import NavDrawer from '../NavDrawer/NavDrawer';
+import CustomSnackbar from '../CustomSnackbar/CustomSnackbar';
+import DeleteConfirmationDialog from '../DeleteConfirmationDialog/DeleteConfirmationDialog';
+
+const VideoCardPage = ({
+  renderSnackbars,
+  title,
+  size,
+  showNavDrawer,
+  videosInformation,
+  showUploadModal,
+  showDeletionDialog,
+  selectedCardIds,
+  showDeletionError,
+  iconsToShow,
+  iconsToShowOnSelected,
```

Comment:
Wth is this monstrosity??

Reply:
Yeah, so, I didn't know how to accomplish not repeating the same code in two places. Any suggestions how to do that, for the future? I mean, how would you think you woulda done it here. Just leave the two extremely similar containers as-is, or pull something out

Reply:
Added spinner